pkg/sink/pulsar.go

```diff
@@ -126,6 +126,11 @@ func (p *PulsarSink) Apply(changes chan source.Change) chan cursor.Checkpoint {
 			return nil
 		}
 
+		if change.Message.GetKeepAlive() != nil {
+			committed <- change.Checkpoint
```

Comment:
Since we are using the `p.producer.SendAsync` below, we can't commit the keepalive checkpoint here or it will be committed out of order.

Reply:
In that case, maybe we can flush all pending request before commit keepalive message
```go
if change.Message.GetKeepAlive() != nil {
	if err := p.producer.Flush(); err != nil {
		return err
	}
	committed <- change.Checkpoint
	return nil
}
```

Reply:
Great, that will work!

Reply:
fixed in: [193fedc](https://github.com/replicase/pgcapture/pull/55/commits/193fedc2e3fde0be2d066217ce2ae94692c7b7be)